core/net/nbr-table.c

```diff
@@ -179,39 +188,66 @@ nbr_table_allocate(void)
   key = memb_alloc(&neighbor_addr_mem);
   if(key != NULL) {
     return key;
-  } else { /* No more space, try to free a neighbor.
-            * The replacement policy is the following: remove neighbor that is:
-            * (1) not locked
-            * (2) used by fewest tables
-            * (3) oldest (the list is ordered by insertion time)
-            * */
-    /* Get item from first key */
-    key = list_head(nbr_table_keys);
-    while(key != NULL) {
-      int item_index = index_from_key(key);
-      int locked = locked_map[item_index];
-      /* Never delete a locked item */
-      if(!locked) {
-        int used = used_map[item_index];
-        int used_count = 0;
-        /* Count how many tables are using this item */
-        while(used != 0) {
-          if((used & 1) == 1) {
-            used_count++;
-          }
+  } else {
+#ifdef NBR_TABLE_FIND_REMOVABLE
+    const linkaddr_t *lladdr;
+    lladdr = NBR_TABLE_FIND_REMOVABLE(reason, data);
+    if(lladdr == NULL) {
+      /* Nothing found that can be deleted - return NULL to indicate failure */
+      PRINTF("*** Not removing entry to allocate new\n");
+      return NULL;
+    } else {
+      /* used least_used_key to indicate what is the least useful entry */
+      int index;
+      int locked;
+      if((index = index_from_lladdr(lladdr)) != -1) {
+        least_used_key = key_from_index(index);
+        locked = locked_map[index];
+      }
+      /* Allow delete of locked item? */
```

Comment:
seems ok to me to allow deletion of a locked item (but I'd remove the question mark for the comment!)